apps/base-docs/docs/building-with-base-and-foundry/testing.md

```diff
@@ -0,0 +1,220 @@
+keywords: [Foundry, Forge, Foundry Book, smart contract development, toolchain, testing, test]
+hide_table_of_contents: false
+---
+
+In this article, you'll learn how to test your smart contracts using [Foundry], the toolchain for smart contract development.
+
+---
+
+## Objectives
+
+By the end of this guide, you should be able to:
+
+- Understand the importance of testing in smart contract development.
+- Write and execute tests for smart contracts using the Forge Standard Library with Foundry.
+- Use some of the `cheatcodes` that Foundry provides.
+
+## Overview
+
+Testing is a crucial aspect of smart contract development, ensuring the reliability and security of your code. Foundry provides a robust testing framework that allows developers to create comprehensive test suites for their projects using Solidity.
+
+## My First Test with Foundry
+
+Considering the default test that the `forge init hello_foundry_in_base` command provides in the seed Foundry project.
+
+```javascript
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import "forge-std/Test.sol";
+import "../src/Counter.sol";
+
+contract CounterTest is Test {
+    Counter public counter;
+
+    function setUp() public {
+        counter = new Counter();
+        counter.setNumber(0);
+    }
+
+    function testIncrement() public {
+        counter.increment();
+        assertEq(counter.number(), 1);
+    }
+
+    function testSetNumber(uint256 x) public {
+        counter.setNumber(x);
+        assertEq(counter.number(), x);
+    }
+}
+```
+
+We can identify the following:
+
+- Test files in Foundry are named following the pattern: <ContractName>.t.sol.
+- Test files are smart contracts named following the pattern: <ContractName>Test.
+- All the tests inherit from `forge-std/Test.sol`.
+- All the tests contain a public function called `setUp`, which is executed before each test. This is similar to the `beforeEach` hook in the Mocha/Typescript world.
+- Test cases start with the `test` keyword, for instance `testIncrement`.
+- Test cases functions are public.
+
+For more information about writing tests in Foundry, you can follow the official guide for [Writing tests]
+
+In order to run the test in Foundry, we simply run:
+
+```bash
+$ forge test
+```
+
+And we should see in the terminal:
+
+```bash
+Running 2 tests for test/Counter.t.sol:CounterTest
+[PASS] testIncrement() (gas: 28334)
+[PASS] testSetNumber(uint256) (runs: 256, μ: 27565, ~: 28343)
+Test result: ok. 2 passed; 0 failed; finished in 13.57ms
+```
+
+## Using Cheatcodes
+
+Foundry includes a set of [Cheatcodes], which are special instructions that are accessible using the `vm` instance in your tests. Cheatcodes allows you to perform various tasks, including:
+
+- Manipulate the state of the blockchain
+- Test reverts
+- Test events
+- Change block number
+- Change identity 
+- And more !
+
+Let's use our first Cheatcode to modify the `msg.sender` of our tests. 
```

Comment:
Again, is it you or we? Because "let us use our" is we. Did someone else write this page? I won't correct the page but you might consider keep the voice the same as the other pages.